typescript-carousel/main.ts

```diff
@@ -0,0 +1,98 @@
+const images: string[] = [
+  './images/001.png',
+  './images/004.png',
+  './images/007.png',
+  './images/025.png',
+  './images/039.png',
+];
+
+let currNdx: number = 0;
+
+const $displayImage = document.querySelector(
+  '.display-image'
+) as HTMLImageElement;
+if (!$displayImage) throw new Error('.display-image query failed!');
+
+$displayImage.setAttribute('src', images[currNdx]);
+
+const $carouselContainer = document.querySelector(
+  '.carousel-container'
+) as HTMLDivElement;
+if (!$carouselContainer) throw new Error('.carousel-container query failed!');
+
+const $leftArrow = document.querySelector('.fa-chevron-left');
+if (!$leftArrow) throw new Error('.fa-chevron-left query failed!');
+const $rightArrow = document.querySelector('.fa-chevron-right');
+if (!$rightArrow) throw new Error('.fa-chevron-right query failed!');
+
+const $circles = document.querySelectorAll('.fa-circle');
+if (!$circles) throw new Error('.fa-circles query failed!');
+
+$carouselContainer.addEventListener('click', function (event: Event) {
+  if (event.target === $leftArrow) {
+    moveLeft();
+  }
+
+  if (event.target === $rightArrow) {
+    moveRight();
+  }
+
+  const element = event.target as HTMLElement;
+  if (element.classList.contains('fa-circle')) {
+    const newNdx: number = parseInt(element.className.split(' ')[0].charAt(5));
+
```

Comment:
That way you can just get the value from the attribute and turn it into a number rather than all of this where you're splitting the `className` property and assuming that the character you want is at character index 5. If you had a double digit number of images this wouldn't work.